src/main/java/com/karankumar/bookproject/backend/service/PublisherService.java

```diff
@@ -0,0 +1,59 @@
+package com.karankumar.bookproject.backend.service;
+
+import java.util.List;
+import java.util.logging.Level;
+
+import org.springframework.stereotype.Service;
+
+import com.karankumar.bookproject.backend.entity.Author;
+import com.karankumar.bookproject.backend.entity.Publisher;
+import com.karankumar.bookproject.backend.repository.PublisherRepository;
+
+import lombok.NonNull;
+import lombok.extern.java.Log;
+
+@Service
+@Log
+public class PublisherService {
+    
+    private final PublisherRepository publisherRepository;
+    
+    public PublisherService(PublisherRepository publisherRepository) {
+        this.publisherRepository = publisherRepository;
+    }
+    
+    public Publisher findById(Long id) {
+        return publisherRepository.getOne(id);
+         
+    }
+    
+    public void save(@NonNull Publisher publisher) {
+        if (publisher.getId() != null) {
+            Publisher existingPublisher = findById(publisher.getId());
+            if (existingPublisher != null) {
+                LOGGER.log(Level.INFO, "Matching PublisherIds: " + existingPublisher.getId());
+                publisher = new Publisher(publisher.getName());
+            }
+        }
```

Comment:
Sorry, I didn't understand why do we have this logic?

Reply:
Its just like Author validation

Reply:
Sorry I might be missing something here, do we really need to copy paste business logic from Author? Sorry if I have missed it and if it was part of the requirement.